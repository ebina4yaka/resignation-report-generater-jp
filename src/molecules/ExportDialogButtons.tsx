import React, { useContext } from 'react'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles'
import Button from '@material-ui/core/Button'
import clsx from 'clsx'
import dayjs from 'dayjs'
import exportDocx from '../libs/export/exportDocx'
import { ExportParameters } from '../libs/export/ExportParameters'
import { parametersContext } from '../context/useParameters'
import convertDateStringToJapanFormat from '../libs/convertDateStringToJapanFormat'
import calcDateOfRetirement from '../libs/calcDateOfRetirement'

type Props = {
  handleClose: () => void
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'row',
      justifyContent: 'center',
    },
    margin: {
      margin: theme.spacing(2),
    },
    button: {
      width: '14ch',
    },
  })
)

export default function ExportDialogButtons(props: Props): React.ReactElement {
  const { handleClose } = props
  const classes = useStyles()
  const context = useContext(parametersContext)
  const {
    name,
    companyName,
    department,
    representativeDirector,
    dateOfNotification,
    reason,
    daysOfPaidLeaveRemaining,
  } = context

  const exportParameters: ExportParameters = {
    name,
    companyName,
    department,
    representativeDirector,
    dateOfNotification: convertDateStringToJapanFormat(dateOfNotification),
    dateOfRetirement: convertDateStringToJapanFormat(
      calcDateOfRetirement(dateOfNotification, daysOfPaidLeaveRemaining)
    ),
    endDateOfPaidLeave: convertDateStringToJapanFormat(
      dayjs(dateOfNotification)
        .add(daysOfPaidLeaveRemaining, 'day')
        .format('YYYY-MM-DD')
    ),
    reason,
    daysOfPaidLeaveRemaining,
  }
  return (
    <div className={classes.root}>
      <Button
        className={clsx(classes.margin, classes.button)}
        variant="contained"
        color="secondary"
        onClick={handleClose}
      >
        入力へ戻る
      </Button>
      <Button
        className={clsx(classes.margin, classes.button)}
        variant="contained"
        color="primary"
        onClick={() => exportDocx(exportParameters)}
      >
        WORD出力
      </Button>
      <Button
        className={clsx(classes.margin, classes.button)}
        variant="contained"
        color="primary"
        onClick={handleClose}
      >
        PDF出力
      </Button>
    </div>
  )
}
