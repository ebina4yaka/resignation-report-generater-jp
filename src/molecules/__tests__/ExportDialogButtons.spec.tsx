import React from 'react'
import renderer from 'react-test-renderer'
import ExportDialogButtons from '../ExportDialogButtons'

test('Snapshot test for ExportDialogButtons', () => {
  const tree = renderer
    .create(<ExportDialogButtons handleClose={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
