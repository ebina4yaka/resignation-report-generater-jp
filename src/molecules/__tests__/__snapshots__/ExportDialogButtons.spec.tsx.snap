// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for ExportDialogButtons 1`] = `
<div
  className="makeStyles-root-1"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-margin-2 makeStyles-button-3 MuiButton-containedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      入力へ戻る
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-margin-2 makeStyles-button-3 MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      WORD出力
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-margin-2 makeStyles-button-3 MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      PDF出力
    </span>
  </button>
</div>
`;
